src/NuGet.Core/NuGet.ProjectModel/Utf8JsonReaderExtensions.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text.Json;
+
+namespace NuGet.ProjectModel
+{
+    internal static class Utf8JsonReaderExtensions
+    {
+        internal static string ReadTokenAsString(this ref Utf8JsonReader reader)
+        {
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.True:
+                    return bool.TrueString;
+                case JsonTokenType.False:
+                    return bool.FalseString;
+                case JsonTokenType.Number:
+                    if (reader.TryGetInt16(out short shortValue))
+                    {
+                        return shortValue.ToString();
+                    }
+                    if (reader.TryGetInt32(out int intValue))
+                    {
+                        return intValue.ToString();
+                    }
+                    else if (reader.TryGetInt64(out long longValue))
+                    {
+                        return longValue.ToString();
+                    }
```

Comment:
Today I'm feeling like while it would technically be faster, maybe the benefit isn't worth it. Especially since there seem to be complications about getting it technically correct, since we'd need to handle when `HasValueSequence` is both true and false, possibly in addition to `ValueIsEscaped`.  So, let's go ahead with the number parsing plus `ToString`.  If perf traces show this `ReadTokenAsString` method taking too much time, then we can consider an optimization.

Something else you could consider, create sub-methods:

```csharp
switch (reader.TokenType)
{
    case JsonTokenType.Number:
        return ReadNumberAsString(ref reader);
    case JsonTokenType.String:
        return ReadStringAsString(ref reader);
   // other cases
}
```

This way the call stack will tell us which token types are actually being hit and taking up the CPU time.